content/blog/2021-09-07-road-to-hell.md

```diff
@@ -0,0 +1,145 @@
+  learning projects will streamline ML and AI development and keep all of us
+  off the road to hell.
+
+descriptionLong: |
+  Our philosophy on how ML and AI should be developed.  Why we believe extending 
+  the best practices of software engineering to the machine learning space will 
+  ultimately be the most effective solution for MLOps.
+
+picture: 2021-09-07/hell.png
+pictureComment: Let's avoid hell, shall we?
+author: dmitry_petrov
+commentsUrl: https://discuss.dvc.org/t/the-road-to-hell-starts-with-good-mlops-intentions/873
+tags:
+  - MLOps
+  - DataOps
+  - CI/CD
+---
+
+Machine learning operations (MLOps) in the last year has emerged as a distinct
+IT discipline for building machine learning (ML) or artificial intelligence (AI)
+models. While at first blush that may seem like a viable method for automating
+the building of AI models, in reality purveyors of MLOps platforms have a vested
+interest in convincing organizations to acquire platforms that exist outside of
+best DevOps practices that have already been proven to accelerate application
+development.
+
+AI models, however, are ultimately a software artifact like any other that needs
+to be integrated within an application. The trouble with MLOps as it is most
+often pursued today is data scientists are constructing AI models in almost
+complete isolation from the rest of the organization. The hope is that somehow
+when the AI model is completed it will magically be incorporated into an
+application development workflow. Unfortunately for all concerned, the rate at
+which applications are being developed using best DevOps practices rarely align
+with the rate at which AI models are being constructed.
+
+> "The trouble with MLOps as it is most often pursued today is data scientists
+> are constructing AI models in almost complete isolation from the rest of the
+> organization."
+
+The result is not only a lot of wasted time and effort, the rate at which
+digital business transformation initiatives that depend on AI models are rolled
+out becomes a significant competitive disadvantage. In effect, the road to AI
+hell is paved with good MLOps intentions.
+
+While working as a data scientist at Microsoft, I saw firsthand how machine
+learning and AI was traditionally implemented in an isolated fashion. That
+unsatisfactory experience led to the launch of opensource Data Version Control
+(DVC) and Continuous Machine Learning (CML) tools that integrate ML workflows
+into best practices for software development. Instead of creating a separate
+proprietary AI platform that needs to be acquired and maintained, the goal needs
+to be to extend traditional software tools such as Git, collaboration and
+continuous integration/continuous delivery (CI/CD) platforms to meet the needs
+of both developers and ML engineers. The entire ML stack needs to be reinvented
+in a way that makes it accessible to every developer.
+
+![DVC Studio](/uploads/images/2021/09-07/dvc-studio.png)
```

Comment:
link is wrong i think?

Reply:
I swear I have the cheese touch @shcheklein 😞

Reply:
it's a typo I think?

`uploads/images/2021/09-07/dvc-studio.png` -> `uploads/images/2021-09-07/dvc-studio.png`

Reply:
Ugh.  The /..... 🤦‍♀️  One moment please....